pkg/upgrade/plan/plan.go

```diff
@@ -117,7 +117,10 @@ func ResolveChannel(ctx context.Context, url, latestVersion, clusterID string) (
 		}
 		return filepath.Base(redirect.Path), nil
 	}
-	return filepath.Base(url), nil
+	if response.StatusCode < 300 {
```

Comment:
I am not sure if we should select any response code less than 300, this is too general

Reply:
i think it's okay if the response is in 200 range except for 302 which is redirected.  
as per our conversation @dweomer

Reply:
Maybe something like `if (response.StatusCode / 200) == 1 {/* return the base of the passed url */}`m what do you think @galal-hussein ?

Reply:
yeah sounds good, so basically limiting the responses to 200 and 300